### PR TITLE
Unify  the errorcode

### PR DESCRIPTION
**Pull request by nanjiek**
### **User description**
What this PR does:

Replaces all instances of erroresponse with errResponse, ensuring a unified error handling mechanism.
Implements the ErrorBuilder pattern for each error type, providing a standardized error response structure.
Adds support for attaching original error information, improving debugging and logging.
Which issue(s) this PR fixes:

Resolves issue [https://github.com/apache/dubbo-go-pixiu/issues/755](https://github.com/apache/dubbo-go-pixiu/issues/755), unifying error codes in the Pixiu project and improving consistency.
Special notes for your reviewer:

Please verify the new error response mechanism, especially the handling of attached error information.

Review the changes related to the unified error codes and make sure they are consistent across the project.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Unified error response mechanism with `ErrorBuilder` pattern across all filters

- Replaced inconsistent error handling with standardized `ErrorResponse` structure

- Added comprehensive test coverage for error response functionality

- Removed deprecated error constants and improved error context with wrapped errors

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Old Error Handling"] --> B["ErrorBuilder Pattern"]
  B --> C["Standardized ErrorResponse"]
  C --> D["JSON Serialization"]
  C --> E["HTTP Status Codes"]
  B --> F["Predefined Error Types"]
  F --> G["404/400/500/503 etc."]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>20 files</summary><table>
<tr>
  <td><strong>erroresponse.go</strong><dd><code>Add ErrorBuilder pattern and ErrorResponse structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-427911168ffa852ef2d4b39a7983030694be206a72d353570869780e3d4afa8f">+141/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>context.go</strong><dd><code>Remove deprecated ErrResponse struct</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-d33e860dc7e2c07b95ec0c928c2a95a3df1b82c22350f559efad76ed5acc6e6e">+0/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>filter.go</strong><dd><code>Remove deprecated default error body constants</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-c1cacef076033b4de6ea4ec57448d912bb9ae9677835ed822eb300b5bc0f2ea0">+0/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>manager.go</strong><dd><code>Update error handling to use ErrorBuilder pattern</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-084e3bd18511b9fe5fccf149b2932043c90ea2b60b8b044231b48f4b235de01e">+4/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>api_config.go</strong><dd><code>Replace error responses with ErrorBuilder pattern</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-e91cca4adadd5ec3751c31e8e1b1b921b2b2537f717f00d5be97dc9775c01cca">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>dubbo.go</strong><dd><code>Unify error handling with ErrorBuilder and improve error context</code></dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-eac58455c0a43fcdd60c5315666741747a95d43d79d839438c853072a411bf65">+19/-18</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>grpc.go</strong><dd><code>Replace error responses with ErrorBuilder pattern</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-ddb337191d25183c0a03323062876cf94c374c17167f3f1dcea1ebf08bf77837">+20/-10</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>routerfilter.go</strong><dd><code>Update error handling to use ErrorBuilder pattern</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-a8b86c386eb00f625a2807472fd846ea8f0c9fecd013ea79df567c707df12266">+8/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>call.go</strong><dd><code>Replace error responses with ErrorBuilder pattern</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-7d79f427e1616490d12457f80993be86fb8577aa4781602c8a64328a5061e250">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>jwt.go</strong><dd><code>Update JWT authentication error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-94f0a5f3934665494c8cc2084f743961056a15d9b3e574962380b8d187045d6a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>authority.go</strong><dd><code>Replace authority filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-24dc3b9f1ddc8f7e0e589453bbd92b2ad8cbf6ba5dad3995064a5fec6e592f83">+2/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>csrf.go</strong><dd><code>Update CSRF filter error responses</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-dd67f7422864aa50fb32280176cc47f78a036eeefc67e9e9adcf60eda2038f66">+4/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>event.go</strong><dd><code>Replace event filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-f11635b1b152b09b8cb86259dc902c41a9c4300cc6747c80de4645a4f257a981">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>filter.go</strong><dd><code>Update LLM proxy filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-37d19eb2e8a72025536b0a071b33558ce790bee9fbe05d43c62c7b7c5194b10a">+6/-10</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>filter.go</strong><dd><code>Replace MCP server filter error responses</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-43bef22a883f33fd267afc7e46cb83e95625298996531f7e692cbaa0e6d51167">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>response.go</strong><dd><code>Update MCP response error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-60d038544338edc80f3c82864fddc3f00153ba14e6fcaad377da839351781529">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>metric.go</strong><dd><code>Replace Prometheus filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-4ddc37c3dfcb2672088ad1c7d5cd28bdc6c392766d52cffb34fc81d05161b8af">+8/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>circuit_breaker.go</strong><dd><code>Update circuit breaker error responses</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-0a8f3440b2de2439217b0ca5763b0cde0feaefc3c870c5c1a5e6bd487556c04f">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>rate_limit.go</strong><dd><code>Replace rate limit filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-02c0cf9813c0fc6b6b418900365ab3d71474b738dceb4e9e600ce22662b0a79d">+2/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>routerfilter.go</strong><dd><code>Update mock router filter error handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-c56d5fcffcf6fc816afb422cbea38cdae24db805322418640a98b13c344f7a71">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>context_test.go</strong><dd><code>Add comprehensive tests for error response functionality</code>&nbsp; </dd></td>
  <td><a href="https://github.com/dubbo-go-pixiu/dubbo-go-pixiu/pull/10/files#diff-9d26a5c3f63492d18f571c8a7f7b50f90d1beef699f7ae24dfefe3bff60254e0">+399/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

- **New Features**
  - Unified structured JSON error responses with consistent status, message, and optional wrapped error details.

- **Refactor**
  - Replaced ad-hoc plain-text and handcrafted JSON replies across proxies, filters, and middleware with standardized error builders and Send/ToJSON flows.

- **Breaking Change**
  - Removed legacy plain-text default error bodies and the old simple error-response type; callers must adopt the new error builders.

- **Tests**
  - Added comprehensive tests covering error builders, serialization, formatting, and edge cases.

### CODE REVIEW

File: pkg/context/http/erroresponse.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		Message string `json:"message"`
+		Error   string `json:"error,omitempty"`
+	}
+	payload := alias{Status: e.Status, Message: e.Message}
+	if e.Err != nil {
+		payload.Error = e.Err.Error()
+	}
+	data, err := json.Marshal(payload)
+	if err != nil {
+		// 避免二次序列化风险，返回静态错误
```
Comment (Alanxtl) [inline pkg/context/http/erroresponse.go]: no Chinese

File: pkg/filter/csrf/csrf.go [hunk 2]
```diff
@@ -90,16 +88,16 @@ func (f *Filter) Decode(ctx *http.HttpContext) filter.FilterStatus {
 	salt := ctx.Request.Header.Get(csrfSalt)
 
 	if salt == "" {
-		bt, _ := json.Marshal(http.ErrResponse{Message: f.cfg.ErrorMsg})
-		ctx.SendLocalReply(stdHttp.StatusForbidden, bt)
+		errResp := http.Forbidden.WithError(fmt.Errorf(f.cfg.ErrorMsg))
```
Comment (Alanxtl) [inline pkg/filter/csrf/csrf.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format
```suggestion
		errResp := http.Forbidden.WithError(errors.New(f.cfg.ErrorMsg))
```

File: pkg/filter/csrf/csrf.go [hunk 3]
```diff
@@ -90,16 +88,16 @@ func (f *Filter) Decode(ctx *http.HttpContext) filter.FilterStatus {
...
+		ctx.SendLocalReply(errResp.Status, errResp.ToJSON())
 		return filter.Stop
 	}
 
 	token := tokenize(f.cfg.Secret, salt)
 
 	if token != tokenGetter(ctx, f.cfg.Key) {
-		bt, _ := json.Marshal(http.ErrResponse{Message: f.cfg.ErrorMsg})
-		ctx.SendLocalReply(stdHttp.StatusForbidden, bt)
+		errResp := http.Forbidden.WithError(fmt.Errorf(f.cfg.ErrorMsg))
```
Comment (Alanxtl) [inline pkg/filter/csrf/csrf.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format
```suggestion
		errResp := http.Forbidden.WithError(errors.New(f.cfg.ErrorMsg))
```

File: pkg/filter/http/apiconfig/api_config.go [hunk 4]
```diff
@@ -116,14 +116,16 @@ func (f *Filter) Decode(ctx *contexthttp.HttpContext) filter.FilterStatus {
...
+		errResp := contexthttp.APINotFound.New()
+		ctx.SendLocalReply(errResp.Status, errResp.ToJSON())
 		e := errors.Errorf("Requested URL %s not found", req.URL.Path)
 		logger.Debug(e.Error())
 		return filter.Stop
 	}
 
 	if !v.Enable {
-		ctx.SendLocalReply(http.StatusNotAcceptable, constant.Default406Body)
+		errResp := contexthttp.NotAcceptable.WithError(fmt.Errorf("API not online"))
```
Comment (Alanxtl) [inline pkg/filter/http/apiconfig/api_config.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		errResp := contexthttp.NotAcceptable.WithError(errors.New("API not online"))
```

File: pkg/filter/http/dubboproxy/dubbo.go [hunk 5]
```diff
@@ -125,8 +124,8 @@ func (f *Filter) Decode(hc *pixiuHttp.HttpContext) filter.FilterStatus {
 
 	if len(splits) != 3 {
 		logger.Info("[dubbo-go-pixiu] http path pattern error. path pattern should be http://127.0.0.1/{application}/{service}/{method}")
-		bt, _ := json.Marshal(pixiuHttp.ErrResponse{Message: "http path pattern error"})
-		hc.SendLocalReply(http.StatusBadRequest, bt)
+		errResp := pixiuHttp.BadRequest.WithError(fmt.Errorf("http path pattern error"))
```
Comment (Alanxtl) [inline pkg/filter/http/dubboproxy/dubbo.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		errResp := pixiuHttp.BadRequest.WithError(errors.New("http path pattern error"))
```

File: pkg/filter/http/dubboproxy/dubbo.go [hunk 6]
```diff
@@ -113,8 +112,8 @@ func (f *Filter) Decode(hc *pixiuHttp.HttpContext) filter.FilterStatus {
 	endpoint := clusterManager.PickEndpoint(clusterName, hc)
 	if endpoint == nil {
 		logger.Info("[dubbo-go-pixiu] cluster not found endpoint")
-		bt, _ := json.Marshal(pixiuHttp.ErrResponse{Message: "cluster not found endpoint"})
-		hc.SendLocalReply(http.StatusServiceUnavailable, bt)
+		errResp := pixiuHttp.ServiceUnavailable.WithError(fmt.Errorf("endpoint not found"))
```
Comment (Alanxtl) [inline pkg/filter/http/dubboproxy/dubbo.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		errResp := pixiuHttp.ServiceUnavailable.WithError(errors.New("endpoint not found"))
```

File: pkg/filter/http/grpcproxy/grpc.go [hunk 7]
```diff
@@ -189,7 +189,8 @@ func (f *Filter) Decode(c *http.HttpContext) filter.FilterStatus {
 	e := server.GetClusterManager().PickEndpoint(re.Cluster, c)
 	if e == nil {
 		logger.Errorf("%s err {cluster not exists}", loggerHeader)
-		c.SendLocalReply(stdHttp.StatusServiceUnavailable, []byte("cluster not exists"))
+		errResp := http.ServiceUnavailable.WithError(fmt.Errorf("cluster not exists"))
```
Comment (Alanxtl) [inline pkg/filter/http/grpcproxy/grpc.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		errResp := http.ServiceUnavailable.WithError(errors.New("cluster not exists"))
```

File: pkg/filter/http/httpproxy/routerfilter.go [hunk 8]
```diff
@@ -119,8 +118,8 @@ func (f *Filter) Decode(hc *contexthttp.HttpContext) filter.FilterStatus {
 	endpoint := clusterManager.PickEndpoint(clusterName, hc)
 	if endpoint == nil {
 		logger.Debugf("[dubbo-go-pixiu] cluster not found endpoint")
-		bt, _ := json.Marshal(contexthttp.ErrResponse{Message: "cluster not found endpoint"})
-		hc.SendLocalReply(http.StatusServiceUnavailable, bt)
+		errResp := contexthttp.ServiceUnavailable.WithError(fmt.Errorf("endpoint not found"))
```
Comment (Alanxtl) [inline pkg/filter/http/httpproxy/routerfilter.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		errResp := contexthttp.ServiceUnavailable.WithError(errors.New("endpoint not found"))
```

File: pkg/filter/llm/proxy/filter.go [hunk 9]
```diff
@@ -150,14 +149,14 @@ func (factory *FilterFactory) PrepareFilterChain(ctx *contexthttp.HttpContext, c
 func (f *Filter) Decode(hc *contexthttp.HttpContext) filter.FilterStatus {
 	rEntry := hc.GetRouteEntry()
 	if rEntry == nil {
-		sendJSONError(hc, http.StatusBadRequest, "no route entry found for request")
+		contexthttp.BadRequest.SendError(hc, fmt.Errorf("no route entry found for request"))
```
Comment (Alanxtl) [inline pkg/filter/llm/proxy/filter.go]: if the format func is not used when creating a new err, use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` only used when need format

```suggestion
		contexthttp.BadRequest.SendError(hc, errors.New("no route entry found for request"))
```

File: pkg/filter/llm/proxy/filter.go [hunk 10]
```diff
@@ -371,8 +370,5 @@ func getNextFallbackEndpoint(currentEndpoint *model.Endpoint, executor *RequestE
 	return nextEndpoint
 }
 
-// sendJSONError is a helper to send a structured JSON error message.
-func sendJSONError(hc *contexthttp.HttpContext, code int, message string) {
-	bt, _ := json.Marshal(contexthttp.ErrResponse{Message: message})
-	hc.SendLocalReply(code, bt)
-}
+// Deprecated: sendJSONError is replaced by ErrorBuilder.SendError
+// Use contexthttp.<ErrorType>.SendError(hc, err) instead
```
Comment (Alanxtl) [inline pkg/filter/llm/proxy/filter.go]: just delete these
```suggestion
```

**Review by Alanxtl — CHANGES_REQUESTED** [review]

File: pkg/common/http/manager.go [hunk 11]
```diff
@@ -101,7 +101,8 @@ func (hcm *HttpConnectionManager) handleHTTPRequest(c *pch.HttpContext) {
 	defer func() {
 		if err := recover(); err != nil {
 			logger.Warnf("[dubbo-go-pixiu] Occur An Unexpected Err: %+v", err)
-			c.SendLocalReply(stdHttp.StatusInternalServerError, []byte(fmt.Sprintf("Occur An Unexpected Err: %v", err)))
+			errResp := pch.InternalError.WithError(fmt.Errorf("panic recovered: %v", err))
+			c.SendLocalReply(errResp.Status, errResp.ToJSON())
```
[the remark this answers is not in the document]
Comment (nanjiek) [reply pkg/common/http/manager.go:105]: 原本这里就是传出错误的，不应该这么修改

[the remark this answers is not in the document]
Comment (nanjiek) [reply pkg/common/http/manager.go:105]: 如果当前错误可以用已经准备好的message就使用new(),如果message过于宽泛，使用witherror()

File: pkg/context/http/erroresponse.go [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+func (eb *ErrorBuilder) SendError(ctx *HttpContext, err error) {
+	errResp := eb.WithError(err)
+	ctx.SendLocalReply(errResp.Status, errResp.ToJSON())
+}
+
+// Send is a convenience method to send error response without error detail
+func (eb *ErrorBuilder) Send(ctx *HttpContext) {
+	errResp := eb.New()
+	ctx.SendLocalReply(errResp.Status, errResp.ToJSON())
+}
```
Comment (Alanxtl) [inline pkg/context/http/erroresponse.go]: do not warp `SendLocalReply` with `Send` and `SendError`, this abstraction is adding to the cognitive load.

**Review by Alanxtl — APPROVED** [review]
lgtm

**Comment by mark4z** [comment]
Nice job! LGTM

**Review by mark4z — APPROVED** [review]
